src/ensembl/src/content/app/browser/track-panel/track-panel-modal/modal-views/TrackPanelBookmarks.tsx

```diff
@@ -4,61 +4,150 @@ import { connect } from 'react-redux';
+import { TrackStates } from 'src/content/app/browser/track-panel/trackPanelConfig';
+import { getActiveGenomePreviouslyViewedObjects } from 'src/content/app/browser/track-panel/trackPanelSelectors';
 import { fetchExampleEnsObjects } from 'src/ens-object/ensObjectActions';
 import { getExampleEnsObjects } from 'src/ens-object/ensObjectSelectors';
 import * as urlFor from 'src/shared/helpers/urlHelper';
 import { getFormattedLocation } from 'src/shared/helpers/regionFormatter';
+import { closeTrackPanelModal } from '../../trackPanelActions';
 
 import upperFirst from 'lodash/upperFirst';
 
 import styles from '../TrackPanelModal.scss';
+import { Bookmark } from '../../trackPanelState';
 
-type TrackPanelBookmarksProps = {
+type StateProps = {
   activeGenomeId: string | null;
-  genomeInfo: GenomeInfoData;
   exampleEnsObjects: EnsObject[];
+  previouslyViewedObjects: Bookmark[];
+};
+
+type DispatchProps = {
   fetchExampleEnsObjects: (objectId: string) => void;
+  updateTrackStates: (trackStates: TrackStates) => void;
+  closeTrackPanelModal: () => void;
+};
+export type TrackPanelBookmarksProps = StateProps & DispatchProps;
+
+const getExampleObjLabel = (exampleObject: EnsObject | Bookmark) => {
+  if (exampleObject.object_type === 'gene') {
+    return exampleObject.label;
+  } else {
+    return getFormattedLocation(exampleObject.location);
+  }
+};
+
+type ExampleLinksProps = Pick<
+  TrackPanelBookmarksProps,
+  'exampleEnsObjects' | 'activeGenomeId' | 'closeTrackPanelModal'
+>;
+export const ExampleLinks = (props: ExampleLinksProps) => {
+  return (
+    <div>
+      {props.exampleEnsObjects.map((exampleObject) => {
+        const path = urlFor.browser({
+          genomeId: props.activeGenomeId,
+          focus: exampleObject.object_id
+        });
+
+        return (
+          <dd key={exampleObject.object_id}>
+            <Link to={path} onClick={props.closeTrackPanelModal}>
+              {getExampleObjLabel(exampleObject)}
+            </Link>
+            <span className={styles.previouslyViewedType}>
+              {' '}
+              {upperFirst(exampleObject.object_type)}
+            </span>
+          </dd>
+        );
+      })}
+    </div>
+  );
+};
+
+type PreviouslyViewedLinksProps = Pick<
+  TrackPanelBookmarksProps,
+  | 'previouslyViewedObjects'
+  | 'updateTrackStates'
+  | 'closeTrackPanelModal'
+  | 'activeGenomeId'
+>;
+
+export const PreviouslyViewedLinks = (props: PreviouslyViewedLinksProps) => {
+  return (
+    <div>
+      {[...props.previouslyViewedObjects]
+        .reverse()
+        .map((previouslyViewedObject, index) => {
+          const locationStr = `${previouslyViewedObject.location.chromosome}:${previouslyViewedObject.location.start}-${previouslyViewedObject.location.end}`;
+          const path = urlFor.browser({
+            genomeId: props.activeGenomeId,
+            focus: previouslyViewedObject.object_id,
+            location: locationStr
+          });
```

Comment:
What we definitely don't want is the location parameter here. We want Genome Browser to take us to the default position of the focus object; and where this default position is, is something that only Genome Browser now knows about. So what we need to pass to Genome Browser is genome id and focus object id. It can be done either by creating React Router links (as here) or by sending direct message to Genome Browser. Either method will work here; I am not sure whether one of them is preferable over the other.

Reply:
> We want Genome Browser to take us to the default position of the focus object;

I initially thought that is the case but we are taking the user to the location they previously viewed. I will look for a way to do this without adding it to the link. Please let me know if you think of an alternative way to do this.